### PR TITLE
Solvers: ValueError when linear_coeffs is used on pure sp.numbers.Float expressions

### LINKED ISSUE #17774: linear_coeffs fails for pure sp.numbers.Float expressions
The `sympy.solvers.solveset.linear_coeffs` method (and therefore also the `linear_eq_to_matrix` method) throws an unjustified error if the input expression `eq` is a constant numerical value of type `sympy.numbers.Float`. For other built-in numerical types (and implicit conversion of integers) this works as expected:

```python
In [19]: import sympy as sp
    ...: from sympy.solvers.solveset import linear_coeffs
    ...: syms = sp.symbols('x y')
    ...: x, y = syms
In [20]: linear_coeffs(1, *syms)  # works as expected
Out[20]: [0, 0, 1]
In [21]: linear_coeffs(sp.numbers.One(), *syms)  # works as expected
Out[21]: [0, 0, 1]
In [22]: linear_coeffs(sp.numbers.Integer(1), *syms)  # works as expected
Out[22]: [0, 0, 1]
In [23]: linear_coeffs(sp.numbers.Float(1.0), *syms)  # throws ValueError
Traceback (most recent call last):
ValueError: nonlinear term encountered: 1.00000000000000
```

If the expression has a variable dependency, it works as well:
```python
In [24]: linear_coeffs(x * sp.numbers.Float(1.0), *syms)  # works as expected
Out[24]: [1.00000000000000, 0, 0]
```

I traced this back to the `core.numbers.as_coeff_add` method, which contains a `self.is_Rational` condition. Ultimately this keeps the `Float` in the `linear_coeffs` local `terms` variable, which makes the expression look nonlinear.
(if the expression is not a pure number, the `core.expr.Expr.as_coeff_add` method is used, which works as expected).
 
I understand that the `self.is_Rational` condition is probably necessary for other applications, such as simplifications. What would be a good fix for this? An early check for constant expressions in  `linear_coeffs`?

### PR DESCRIPTION
**Pull request by sachin-4099**
Fixes #17774 

> In [19]: import sympy as sp
    ...: from sympy.solvers.solveset import linear_coeffs
    ...: syms = sp.symbols('x y')
    ...: x, y = syms
In [20]: linear_coeffs(1, *syms)  # works as expected
Out[20]: [0, 0, 1]
In [21]: linear_coeffs(sp.numbers.One(), *syms)  # works as expected
Out[21]: [0, 0, 1]
In [22]: linear_coeffs(sp.numbers.Integer(1), *syms)  # works as expected
Out[22]: [0, 0, 1]
In [23]: linear_coeffs(sp.numbers.Float(1.0), *syms)  **# throws ValueError**
Traceback (most recent call last):
**ValueError: nonlinear term encountered: 1.00000000000000**

After the change:
>> In [19]: import sympy as sp
    ...: from sympy.solvers.solveset import linear_coeffs
    ...: syms = sp.symbols('x y')
    ...: x, y = syms
In [20]: linear_coeffs(1, *syms)  # works as expected
Out[20]: [0, 0, 1]
In [21]: linear_coeffs(sp.numbers.One(), *syms)  # works as expected
Out[21]: [0, 0, 1]
In [22]: linear_coeffs(sp.numbers.Integer(1), *syms)  # works as expected
Out[22]: [0, 0, 1]
In [23]: linear_coeffs(sp.numbers.Float(1.0), *syms)  **# works as expected**
**Out[23]: [0, 0, 1.0]**

* solvers
  * linear_coeffs gives no longer raises when expr has no symbols of interest

### CODE REVIEW

**Comment by sachin-4099** [comment]
Any testcase needs to be added @smichr ?

**Comment by sachin-4099** [comment]
If any changes are required please suggest, otherwise the PR is ready to be merged @smichr.

**Comment by czgdp1807** [comment]
Please add an appropriate test for the fix. Thanks.

File: sympy/solvers/tests/test_solveset.py [hunk 1]
```diff
@@ -2092,6 +2095,7 @@ def test_linear_coeffs():
     raises(ValueError, lambda:
         linear_coeffs(1/x*(x - 1) + 1/x, x))
     assert linear_coeffs(a*(x + y), x, y) == [a, a, 0]
+    assert linear_coeffs(sp.numbers.Float(1.0), *syms) == [0, 0, 1.0]
```
Comment (oscarbenjamin) [inline sympy/solvers/tests/test_solveset.py]: You can just use 1.0 rather than `sp.numbers.Float(1.0)`

Comment (smichr) [reply sympy/solvers/tests/test_solveset.py]: And `x` and `y` are defined at the top of the file so the only line needed is `assert linear_coeffs(1.0, x, y) == [0, 0, 1.0]`

Comment (sachin-4099) [reply sympy/solvers/tests/test_solveset.py]: Done the necessary changes.

**Comment by sachin-4099** [comment]
But I am not able to understand what is causing the testcase in test_invert_modular to fail
@smichr @oscarbenjamin

**Comment by sachin-4099** [comment]
If any changes are required please suggest, otherwise the PR is ready to be merged @czgdp1807 @smichr @oscarbenjamin.

**Pull request merged**